File: SGEI-API/SGEI/Models/Security/PermissionxModule.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SGEI.Models.Security
{
  public class PermissionxModule
  {
    public long id { get; set; }

    public int idmodulo { get; set; }

    public int idpermiso { get; set; }

    public bool activo { get; set; }

    [ForeignKey(nameof(idmodulo))]
    public Module modulo { get; set; }

    [ForeignKey(nameof(idpermiso))]
    public Permission permiso { get; set; }
  }
}

File: src/app/models/security/permissionxmodule.ts

import { Module } from "./module";
import { Permissions } from "./permissions";

export class PermissionxModule{
    id: number = -1;
    idmodulo: number = -1;
    idpermiso: number = -1;
    activo: boolean = false;
    permiso: Permissions = new Permissions();
    modulo: Module = new Module();
    checked: boolean = false;
}